pkg/util/test/util.go

```diff
@@ -413,11 +413,45 @@ func UninstallStorageCluster(cluster *corev1.StorageCluster, kubeconfig ...strin
 		if _, err = operatorops.Instance().UpdateStorageCluster(cluster); err != nil {
 			return err
 		}
+
+		defaultCheckReferenceTimeout := 30 * time.Second
+		defaultCheckReferenceInterval := 1 * time.Second
+		if err = checkTelemetrySecretGotOwnerReference(cluster, defaultCheckReferenceTimeout, defaultCheckReferenceInterval); err != nil {
+			return err
+		}
 	}
 
 	return operatorops.Instance().DeleteStorageCluster(cluster.Name, cluster.Namespace)
 }
 
+func checkTelemetrySecretGotOwnerReference(cluster *corev1.StorageCluster, timeout, interval time.Duration) error {
```

Comment:
if telemetry disabled, return nil directly, only check ref when telemetry enabled.

Reply:
Won't do this as secret might exist from previously enabled telemetry